### ISSUE TITLE
run_train.sh example failure

### ISSUE THREAD

**Issue by shaofushih** [PaddlePaddle/DeepSpeech#194]
I kept getting cudnn errors on CUDA 9.1 and CUDNN 7.1 on Ubuntu 16.04
Tried reinstalling paddlepaddle and kept getting the follow erros

-----------  Configuration Arguments -----------
augment_conf_path: conf/augmentation.config
batch_size: 16
dev_manifest: data/tiny/manifest.tiny
init_model_path: None
is_local: 1
learning_rate: 1e-05
max_duration: 27.0
mean_std_path: data/tiny/mean_std.npz
min_duration: 0.0
num_conv_layers: 2
num_iter_print: 100
num_passes: 20
num_proc_data: 1
num_rnn_layers: 3
output_model_dir: ./checkpoints/tiny
rnn_layer_size: 2048
share_rnn_weights: 1
shuffle_method: batch_shuffle_clipped
specgram_type: linear
test_off: 0
train_manifest: data/tiny/manifest.tiny
trainer_count: 4
use_gpu: 1
use_gru: 0
use_sortagrad: 1
vocab_path: data/tiny/vocab.txt
------------------------------------------------
I0327 16:29:22.122421 13135 Util.cpp:166] commandline:  --use_gpu=1 --rnn_use_batch=True --log_clipping=True --trainer_count=4 
F0327 16:29:22.784910 13135 DynamicLoader.cpp:104] Check failed: nullptr != *dso_handle Failed to find dynamic library: libcudnn.so (libcudnn.so: cannot open shared object file: No such file or directory) 
Please specify its path correctly using following ways: 
Method. set environment variable LD_LIBRARY_PATH on Linux or DYLD_LIBRARY_PATH on Mac OS. 
For instance, issue command: export LD_LIBRARY_PATH=... 
Note: After Mac OS 10.11, using the DYLD_LIBRARY_PATH is impossible unless System Integrity Protection (SIP) is disabled.
*** Check failure stack trace: ***
    @     0x7f4be2c74bcd  google::LogMessage::Fail()
    @     0x7f4be2c7867c  google::LogMessage::SendToLog()
    @     0x7f4be2c746f3  google::LogMessage::Flush()
    @     0x7f4be2c79b8e  google::LogMessageFatal::~LogMessageFatal()
    @     0x7f4be2bb3bad  GetCudnnDsoHandle()
    @     0x7f4c5a364a99  __pthread_once_slow
    @     0x7f4be2c27372  hl_cudnn_init()
    @     0x7f4be2c2fd3e  hl_create_global_resources()
    @     0x7f4be2c30439  hl_specify_devices_start()
    @     0x7f4be2c3085d  hl_start()
    @     0x7f4be2bba55a  paddle::initMain()
    @     0x7f4be2c5c5d1  initPaddle()
    @     0x7f4be2803c29  _wrap_initPaddle
    @           0x4c30ce  PyEval_EvalFrameEx
    @           0x4b9ab6  PyEval_EvalCodeEx
    @           0x4c1e6f  PyEval_EvalFrameEx
    @           0x4b9ab6  PyEval_EvalCodeEx
    @           0x4d54b9  (unknown)
    @           0x4a577e  PyObject_Call
    @           0x4bed3d  PyEval_EvalFrameEx
    @           0x4b9ab6  PyEval_EvalCodeEx
    @           0x4c1e6f  PyEval_EvalFrameEx
    @           0x4b9ab6  PyEval_EvalCodeEx
    @           0x4c1e6f  PyEval_EvalFrameEx
    @           0x4b9ab6  PyEval_EvalCodeEx
    @           0x4eb30f  (unknown)
    @           0x4e5422  PyRun_FileExFlags
    @           0x4e3cd6  PyRun_SimpleFileExFlags
    @           0x493ae2  Py_Main
    @     0x7f4c59fac830  __libc_start_main
    @           0x4933e9  _start
    @              (nil)  (unknown)
Aborted (core dumped)
Fail in training!

**Comment by shaofushih** [comment]
Here is my system configuration:

nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2017 NVIDIA Corporation
Built on Fri_Nov__3_21:07:56_CDT_2017
Cuda compilation tools, release 9.1, V9.1.85

Tue Mar 27 16:35:33 2018       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 387.34                 Driver Version: 387.34                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 108...  Off  | 00000000:03:00.0 Off |                  N/A |
|  0%   41C    P8    11W / 280W |      2MiB / 11172MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   1  GeForce GTX 108...  Off  | 00000000:04:00.0  On |                  N/A |
|  0%   54C    P2    55W / 280W |    350MiB / 11169MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    1      1335      G   /usr/lib/xorg/Xorg                           195MiB |
|    1      2110      G   compiz                                       152MiB |
+-----------------------------------------------------------------------------+

**Comment by pkuyym** [comment]
> F0327 16:29:22.784910 13135 DynamicLoader.cpp:104] Check failed: nullptr != *dso_handle Failed to find dynamic library: libcudnn.so (libcudnn.so: cannot open shared object file: No such file or directory)
> Please specify its path correctly using following ways:
> Method. set environment variable LD_LIBRARY_PATH on Linux or DYLD_LIBRARY_PATH on Mac OS.
> For instance, issue command: export LD_LIBRARY_PATH=...
> Note: After Mac OS 10.11, using the DYLD_LIBRARY_PATH is impossible unless System Integrity 

The log is very clear, please make sure path of cudnn libs in LD_LIBRARY_PATH. You can export `LD_LIBRARY_PATH=${your_cudnn_lib_path}:$LD_LIBRARY_PATH`

**Comment by shaofushih** [comment]
Thanks for the quick reply. After manually putting the cudnn files into cuda-9.1, I am now facing the version mismatch issue as the following:

F0328 16:49:17.634479 14139 hl_cuda_cudnn.cc:171] Check failed: (cudnn_cuh_major < 4 && cudnn_dso_major < 4) || (cudnn_cuh_major == cudnn_dso_major) [cudnn init] libcudnn v7 with header v5 unmatched!
PaddlePaddle Requirement: (header v[2-3] with libcudnn v[2-3]) Or (header v4 with libcudnn v4) Or (header v5 with libcudnn v5) Or(header v6 with libcudnn v6).
*** Check failure stack trace: ***
    @     0x7f6694ad9bcd  google::LogMessage::Fail()
    @     0x7f6694add67c  google::LogMessage::SendToLog()
    @     0x7f6694ad96f3  google::LogMessage::Flush()
    @     0x7f6694adeb8e  google::LogMessageFatal::~LogMessageFatal()
    @     0x7f6694a8c479  hl_cudnn_init()
    @     0x7f6694a94d3e  hl_create_global_resources()
    @     0x7f6694a95439  hl_specify_devices_start()
    @     0x7f6694a9585d  hl_start()
    @     0x7f6694a1f55a  paddle::initMain()
    @     0x7f6694ac15d1  initPaddle()
    @     0x7f6694668c29  _wrap_initPaddle
    @           0x4c30ce  PyEval_EvalFrameEx
    @           0x4b9ab6  PyEval_EvalCodeEx
    @           0x4c1e6f  PyEval_EvalFrameEx
    @           0x4b9ab6  PyEval_EvalCodeEx
    @           0x4d54b9  (unknown)
    @           0x4a577e  PyObject_Call
    @           0x4bed3d  PyEval_EvalFrameEx
    @           0x4b9ab6  PyEval_EvalCodeEx
    @           0x4c1e6f  PyEval_EvalFrameEx
    @           0x4b9ab6  PyEval_EvalCodeEx
    @           0x4c1e6f  PyEval_EvalFrameEx
    @           0x4b9ab6  PyEval_EvalCodeEx
    @           0x4eb30f  (unknown)
    @           0x4e5422  PyRun_FileExFlags
    @           0x4e3cd6  PyRun_SimpleFileExFlags
    @           0x493ae2  Py_Main
    @     0x7f670b72a830  __libc_start_main
    @           0x4933e9  _start
    @              (nil)  (unknown)
Aborted (core dumped)

**Comment by shaofushih** [comment]
Hello again,

Does anyone know if cudnn v6 is compatible with cuda 9.0? If not I am afraid to break my current tensorflow installation by rolling back to CUDA 8.0 + cudnn v6.

Thanks in advance.
